code/collections.rb

```diff
@@ -1,15 +1,41 @@
 # collections
 
 # 1. Create a variable `point1` and assign it a hash representing a coordinate with keys `:x` and `:y` which have integer values
+  point1 = {
+    x: 2,
+    y: 4
+  }
 
 # 2. Print the y value
+puts point1[:y]
 
 # 3. Make an array, `points`, that includes `point1` and two more points (`points.length` should be 3)
 
+point2 = {
+  x:1,
+  y:3
+}
+
+point3 = {
+  x:3,
+  y:6
+}
+
+points = [point1, point2, point3]
+
+
 # 4. Use `.each` to print the y value of each point
 
+points.each { |point| puts point[:y] }
+
+
 # 5. Use `.each_with_index` and string interpolation to print for each point "The point at index <index> has coordinates {x: <x-val>, y: <y-val>}"
 
+points.each_with_index { |point, index| puts "the point at index #{index} has coordinates #{point}" }
```

Comment:
You got each point printed with it's index as requested. 👍 But notice ruby `puts` a hash with "hash rockets" for the key-value pairs `{:x=>1, :y=>2}` instead of colons `{x: 1, y: 2}`. If you want to make it look like the question prompt, jus replace the `<x-val>` and `<y-val>` with the corresponding point[:x] and point[:y] like this:

```ruby
puts "The point at index #{index} has coordinates {x: #{point[:x]}, y: #{point[:y]}}"
```